AutoCollection/PreAggregatedMetrics.ts

```diff
@@ -0,0 +1,301 @@
+                        dimensions: currentCounter.dimensions,
+                        value: averageDependencyExecutionTime,
+                        count: intervalDependencies,
+                        aggregationInterval: elapsedMs,
+                        metricType: Constants.MetricId.DEPENDENCIES_DURATION,
+                    });
+                }
+            }
+            // Set last counters
+            currentCounter.lastTotalCount = currentCounter.totalCount;
+            currentCounter.lastTime = currentCounter.time;
+        }
+    }
+
+    private static _trackExceptionMetrics() {
+        for (let i = 0; i < this._exceptionCountersCollection.length; i++) {
+            var currentCounter = this._exceptionCountersCollection[i];
+            var intervalExceptions = (currentCounter.totalCount - currentCounter.lastTotalCount) || 0;
+            var elapsedMs = currentCounter.time - currentCounter.lastTime;
+            this._trackPreAggregatedMetric({
+                name: "Exceptions",
+                dimensions: currentCounter.dimensions,
+                value: intervalExceptions,
+                count: intervalExceptions,
+                aggregationInterval: elapsedMs,
+                metricType: Constants.MetricId.DEPENDENCIES_DURATION,
+            });
+
+            // Set last counters
+            currentCounter.lastTotalCount = currentCounter.totalCount;
+            currentCounter.lastTime = currentCounter.time;
+        }
+    }
+
+    private static _trackTraceMetrics() {
+        for (let i = 0; i < this._traceCountersCollection.length; i++) {
+            var currentCounter = this._traceCountersCollection[i];
+            var intervalTraces = (currentCounter.totalCount - currentCounter.lastTotalCount) || 0;
+            var elapsedMs = currentCounter.time - currentCounter.lastTime;
+            this._trackPreAggregatedMetric({
+                name: "Traces",
+                dimensions: currentCounter.dimensions,
+                value: intervalTraces,
+                count: intervalTraces,
+                aggregationInterval: elapsedMs,
+                metricType: Constants.MetricId.DEPENDENCIES_DURATION,
+            });
+
+            // Set last counters
+            currentCounter.lastTotalCount = currentCounter.totalCount;
+            currentCounter.lastTime = currentCounter.time;
+        }
+    }
+
+    private static _trackPreAggregatedMetric(metric: AggregatedMetric) {
+        // Build metric properties
+        let metricProperties: any = {};
+        for (let dim in metric.dimensions) {
+            metricProperties[PreaggregatedMetricPropertyNames[dim as MetricDimensionTypeKeys]] = metric.dimensions[dim];
+        }
+        metricProperties = {
+            ...metricProperties,
+            "_MS.MetricId": metric.metricType,
+            "_MS.AggregationIntervalMs": String(metric.aggregationInterval),
+            "_MS.IsAutocollected": "True",
```

Comment:
You need to populate `Request.Success` = `True/False` depending on success or not for requests/dependencies failed. This is in ADDITION to `request/success` and `dependencies/success` dimensions.